docs/contribution-guides/Development-Standards-Guide.md

```diff
@@ -226,3 +242,301 @@ to just leave it as a public field without a property. As always, make sure to d
+
+Events can be used to reduce the likelyhood of bugs and [NREs](https://learn.microsoft.com/en-us/dotnet/api/system.nullreferenceexception?view=net-7.0) occuring and makes our code much more maintainable and easy to expand upon due to how they're designed.
+
+With events, components don't have to constantly reference each other and only require a subscription to relevent functions that get invoked when things happen in the game.
+
+```cs
+//example of bad code
+
+class Health : Monobehavior
+{
+    public bool StateChanged;
+    public int currentHealth = 5;
+
+    void stateChange(int damageTaken){
+        currentHealth -= damageTaken;
+        if(currentHealth >= 0) return;
+        StateChanged = true;
+        Inventory.DropAll(); //How do we know that we have an inventory at all?
+    }
+}
+
+class DeathUI : Monobehavior
+{
+    public Health health; //<- possibily null
+
+    void Awake(){
+        UpdateManager.Add(FIXED_UPDATE, myFunc);
+    }
+
+    void myFunc(){
+        if(health.StateChanged == false) return;
+        DoThing();
+    }
+}
+
+class PlayerInventory : Inventory{
+    public void DropAll();
+}
+```
+
+```cs
+//example of good code
+
+class Health : Monobehavior
+{
+    public int CurrentHealth {private set; get;} = 5;
+    public Action OnDeath = new Action();
+
+    public void TakeDamage(int damageTaken)
+    {
+        currentHealth -= damageTaken;
+        if(currentHealth >= 0) return;
+        OnDeath?.Invoke();
+    }
+}
+
+class DeathUI : Monobehavior
+{
+    void OnEnable()
+    {
+        if(gameObject.TryGetComponent<Health>(out var playerHealth) == false) 
+        {
+            // Always leave warnings and errors whenever possible instead of silently not reporting unintended behaviors.
+            Logger.LogWarning("[DeathUI/OnEnable] - No health found on this gameObject! DeathUI hasn't subscribed to any events when it got enabled!");
+            return;
+        }
+        // Subscribes to the OnDeath event which will trigger when this object's health reaches 0 or below.
+        playerHealth.OnDeath += myFunc;
+    }
+
+    void myFunc()
+    {
+        DoThing();
+    }
+}
+
+class PlayerInventory : Inventory
+{
+    //same thing as in DeathUI, just subscribe to the OnDeath event.
+}
+```
+
+_Note: Events can leak! Always remember to unsubscribe your events on `OnDisable()`._
+
+* Minimise Polymorphism, Embrace Interfaces.
+
+[Interface](https://www.w3schools.com/cs/cs_interface.php) in C# is a blueprint of a class. It is like abstract class because all the methods which are declared inside the interface are abstract methods. It cannot have method body and cannot be instantiated. It is used to achieve multiple inheritance which can't be achieved by class.
+
+Interfaces allow us to easily communicate features between objects and managers by keeping design/implamentation contained in predictable states and architecture while allowing us to easily expand, maintain and iterate upon them.
```

Comment:
Allows to share behaviour with classes in a horizontal manner, contrary to inheritance where the sharing is vertical (from parent to children). In horizontal, there is no need for a parental relationship between the classes. Not sure how you got to the "predictable state" part.